scripts/staging/slicing/base/node.py

```diff
@@ -0,0 +1,102 @@
+from init import init
+
+init = init()
+enc = init[0]
+model = init[1]
+complete_x = init[2]
+complete_y = init[3]
+f_l2 = init[4]
+x_size = init[5]
+x_test = init[6]
+y_test = init[7]
+
+
+class Node:
+    name: ""
+    attributes: []
+    parents: []
+    children: []
+    size: int
+    l2_error: float
+    score: float
+    s_upper: int
+    s_lower: int
+    e_upper: float
+    c_upper: float
+
+    def __init__(self):
+        self.parents = []
+        self.attributes = []
+
+    def calc_c_upper(self):
+        upper_score = (self.e_upper / self.s_lower) / (f_l2 / x_size) + self.s_upper
+        return float(upper_score)
+
+    def make_slice(self):
+        subset = []
+        for row in complete_x:
+            flag = True
+            for feature in self.attributes:
+                if row[1][feature[1]] != 1:
+                    flag = False
```

Comment:
Try to vectorize this function if possible as this is used in the inner loop. I could imagine something like 
```
I = (rowSums(complete_x == mask) == numFeatures) # mask being a row vector
subset = select(complete_x, I)
```